packages/contracts/contracts/libs/v0.8.x/minter-libs/DALib.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Art Blocks Merkle Library
+ * @notice This library is designed to manage and verify merkle based gating for Art Blocks projects.
+ * It provides functionalities such as updating the merkle root of project, verifying an address against a proof,
+ * and setting the maximum number of invocations per address for a project.
+ * @author Art Blocks Inc.
+ */
+
+library DALib {
+    using SafeCast for uint256;
+
+    struct DAProjectConfig {
+        uint64 timestampStart;
+        uint64 timestampEnd;
+        uint256 startPrice;
+        uint256 basePrice;
+        uint64 priceDecayHalfLifeSeconds;
+    }
+
+    function resetAuctionDetailsExp(
+        DAProjectConfig storage _auctionProjectConfigMapping
+    ) external {
+        _auctionProjectConfigMapping.timestampStart = 0;
+        _auctionProjectConfigMapping.priceDecayHalfLifeSeconds = 0;
+        _auctionProjectConfigMapping.startPrice = 0;
+        _auctionProjectConfigMapping.basePrice = 0;
+    }
+
+    function resetAuctionDetailsLin(
+        DAProjectConfig storage _auctionProjectConfigMapping
+    ) external {
+        // reset to initial values
+        _auctionProjectConfigMapping.timestampStart = 0;
+        _auctionProjectConfigMapping.timestampEnd = 0;
+        _auctionProjectConfigMapping.startPrice = 0;
+        _auctionProjectConfigMapping.basePrice = 0;
+    }
+
+    function setAuctionDetailsLin(
+        DAProjectConfig storage _auctionProjectConfigMapping,
+        uint256 _auctionTimestampStart,
+        uint256 _auctionTimestampEnd,
+        uint256 _startPrice,
+        uint256 _basePrice,
+        uint256 _minimumAuctionLengthSeconds
```

Comment:
nit: since`_minimumAuctionLengthSeconds` is more of a minter-specific implementation detail, and not as fundamental to the auction logic itself, WDYT about keeping that value and check outside of the library, and instead performing it on the minter. My line of thinking here is "if we were to remove this guard rail in a future iteration of this minter, if we keep the check in the minter, the library would not have to be updated". No super strong opinions here, but interested in your thoughts.

Reply:
agreed, i think that line of thinking is pretty compelling here (avoid the unnecessary path that makes future upgrades more painful)